test/clj/rems/api/test_applications.clj

```diff
@@ -434,31 +434,37 @@
         form-id (create-form-with-fields [{:title {:en "some attachment"}
                                            :type "attachment"
                                            :optional true}])
-        field-id (-> (form/get-form form-id) :items first :id)
         cat-id (create-catalogue-item form-id workflow-id)
         app-id (create-application [cat-id] user-id)
         upload-request (fn [file]
-                         (-> (request :post (str "/api/applications/add-attachment?application-id=" app-id "&field-id=" field-id))
+                         (-> (request :post (str "/api/applications/add-attachment?application-id=" app-id))
                              (assoc :params {"file" file})
                              (assoc :multipart-params {"file" file})))
-        read-request (request :get "/api/applications/attachments" {:application-id app-id :field-id field-id})]
+        read-request #(request :get (str "/api/applications/attachment/" %))]
     (testing "uploading attachment for a draft"
       (let [body (-> (upload-request filecontent)
                      (authenticate api-key user-id)
                      handler
-                     read-ok-body)]
-        (is (= {:success true} body))))
+                     read-ok-body)
+            id (:id body)]
+        (is (:success body))
+        (is (number? id))
+        (testing "and retrieving it as the applicant"
+          (let [response (-> (read-request id)
+                             (authenticate api-key user-id)
+                             handler
+                             assert-response-is-ok)]
+            (is (= (slurp testfile) (slurp (:body response))))))
+        (testing "and retrieving it as non-applicant"
+          (let [response (-> (read-request id)
+                             (authenticate api-key "carl")
+                             handler)]
+            (is (response-is-forbidden? response))))))
```

Comment:
There doesn't seem to be a test for attachment not found (i.e. wrong attachment ID). That would be a good addition to make sure that the `if attachment` code works right.